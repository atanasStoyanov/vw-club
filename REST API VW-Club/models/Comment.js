const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const Model = mongoose.model;
const { String, Number, Boolean, ObjectId } = Schema.Types;

const commentSchema = new Schema({
    comment: {
        type: String,
        required: true
    },
    author: {
        type: ObjectId,
        ref: "User"
    }

});

module.exports = new Model('Comment', commentSchema);