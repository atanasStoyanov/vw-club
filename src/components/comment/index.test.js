import React from 'react';
import renderer from 'react-test-renderer';
import Comment from '.';

describe('Comment Component', () => {
    it('should render comment with text "Test" and author "Pesho"', () => {
        const author = {
            username: 'Pesho',
        }
        const tree = renderer.create(<Comment comment='Test' author={author}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})