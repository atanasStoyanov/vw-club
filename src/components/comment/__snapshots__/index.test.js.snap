// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component should render comment with text "Test" and author "Pesho" 1`] = `
<div
  className="container"
>
  <div
    className="element"
  >
    <img
      alt="avatar"
      src="profile-icon.png"
      style={
        Object {
          "height": "50px",
          "width": "60px",
        }
      }
    />
  </div>
  <div
    className="element"
  >
    <h3>
       
      Pesho
    </h3>
    <div
      className="comment"
    >
      <p>
        Test
      </p>
    </div>
  </div>
</div>
`;
